core/src/test/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapperTests.java

```diff
@@ -472,30 +476,109 @@ public void testLonLatArrayArrayStored() throws Exception {
         assertThat(doc.rootDoc().getFields("point")[1].stringValue(), equalTo("1.4,1.5"));
     }
 
+
+    @Test
+    public void testDeprecatedOptions() throws Exception {
+        DocumentMapperParser parser = createIndex("test").mapperService().documentMapperParser();
+        // test deprecated validate
+        try {
+            String validateMapping = XContentFactory.jsonBuilder().startObject().startObject("type")
+                    .startObject("properties").startObject("point").field("type", "geo_point").field("lat_lon", true).field("geohash", true)
+                    .field("validate", true).endObject().endObject()
+                    .endObject().endObject().string();
+            parser.parse(validateMapping);
+            Assert.fail("process completed successfully when " + MapperParsingException.class.getName() + " expected");
+        } catch (Exception e) {
+            assert(e.getClass().equals(MapperParsingException.class)):
```

Comment:
Why not just catch MapperParsingException? Then you can check the message it has.